source/features/edit-comments-faster.tsx

```diff
@@ -1,14 +1,22 @@
 import React from 'dom-chef';
+import select from 'select-dom';
 import onetime from 'onetime';
 import {observe} from 'selector-observer';
 import {PencilIcon} from '@primer/octicons-react';
 import * as pageDetect from 'github-url-detection';
 
 import features from '.';
 
+function canEditDiscussion(): boolean {
+	return pageDetect.isDiscussion() && select.exists(`
+	.current-user span:is([title^="You are a maintainer on this repository"], [title^="You have been invited to collaborate on"]),
+	button[aria-label="Edit discussion title"]
```

Comment:
Is this not enough? In what case is the other needed? You can use an array of selectors if necessary, and add comments on each selector (see other features for style)

Reply:
This is what you suggest, right?

```js
function canEditDiscussion(): boolean {
	return pageDetect.isDiscussion() && select.exists([
		'.current-user span[title^="You are a maintainer on this repository"]',  // You are the author of the repo
		'.current-user span[title^="You have been invited to collaborate on"]',  // You are a collaborator
		'button[aria-label="Edit discussion title"]' // You can edit the discussion title (being collaborator is not always enough)
	]);
}
```

_I agree that using array is easier to read, than using template literals <code>\`...\`</code> and `is()`._
But, I'd prefer @yakov116 to share whether he is ok with this change.

Reply:
> _combining the two with `:is()`_.

Both can be used, what I meant is that the 2 existing lines can be 2 selectors. `:is()` is unchanged.



> This is what you suggest, right?

The comments don't add much information, they just repeat the selector in English. What I mean is something like "This is for mainainers/collaborators" and "This is for xyz"

What I asked though is _why is the last selector alone not enough?_ What role:

- lets you edit every comment, BUT
- doesn't let you change the title

In my mind, if you can edit every comment, you can also edit the title.

Reply:
Please help me clarify this:

In [here](https://github.com/sindresorhus/refined-github/pull/4415#discussion_r640047844) you said this:

> However I'm not seeing #4400 at all
> [[screenshot](https://user-images.githubusercontent.com/1402241/119717057-0a299f00-be90-11eb-95ac-b2c4692391cf.png)]
> Probably because @cheap-glitch is a collaborator there but can't edit the comments, whereas I'm a stranger  
> `canEditDiscussion` needs to be created with this in mind.

Υou say that one can be a collaborator, but still not edit every comment.
This can happen where? In both Issues and Discussions? 

I'm asking because I have been invited as collaborator in https://github.com/tophf/mpiv and https://github.com/Mottie/GitHub-userscripts
and I can edit all Issue/PR/discussion comments in both repos.

---

If the above is true, i.e. that for some repos, a collaborator cannot edit every comment,
then, making `canEditDiscussion` rely solely on whether you can edit the title, is not reliable: 
you may edit the title, simply because you started the discussion, i.e. you are the author of the discussion.

So, I believe the code should change to the following:

```js
function canEditDiscussion(): boolean {
	return pageDetect.isDiscussion() && 
    ( select.exists('.current-user span[title^="You are a maintainer on this repository"') || 
    ( select.exists('.current-user span[title^="You have been invited to collaborate on"') &&
      select.exists('button[aria-label="Edit discussion title"]') );
}
```

Reply:
> > However I'm not seeing #4400 at all
> > [[screenshot](https://user-images.githubusercontent.com/1402241/119717057-0a299f00-be90-11eb-95ac-b2c4692391cf.png)]
> > Probably because @cheap-glitch is a collaborator there but can't edit the comments, whereas I'm a stranger  
> > `canEditDiscussion` needs to be created with this in mind.
> 
> Υou say that one can be a collaborator, but still not edit every comment.
> This can happen where? In both Issues and Discussions? 

That was complete guesswork. The only thing I said there is that I don't see the bug and I don't know why.

I'm here asking _you_ to verify it. We don't need code based on suppositions that nobody tested 😃

Reply:
I thought you actually noticed this somewhere.

So far I could only verify this in my own test repo, and in those two repos that I am collaborator.

For the sake of testing I made a test comment in https://github.com/streetcomplete/StreetComplete/discussions/2671#discussioncomment-801868
and didn't notice the issue:

<details>
<summary>screenshot:</summary>

![2021-05-29_202249](https://user-images.githubusercontent.com/723651/120079239-ab486d80-c0bb-11eb-854b-2af398ecb661.jpg)
</details>

&nbsp;
Please note that @FloEdelmann is a *member* of that organization ([screenshot](https://user-images.githubusercontent.com/723651/120079348-0da16e00-c0bc-11eb-87a4-a5bf487ea569.jpg)) whereas I am a stranger, 
so there's nothing more I can do in terms of testing.

Reply:
Update:
I noticed that the text in Discussions is different than in Issues and PRs:
it's `You have been invited to collaborate on` on Issues and PRs,
while it's `You are a collaborator on this repository` on Discussions.
I just corrected the selector.

I also removed the `'button[aria-label="Edit discussion title"]'`
because as I said above, is an unreliable criteria:
you may edit the title, simply because you started the discussion, i.e. you are the author of the discussion.